granulate_utils/linux/ns.py

```diff
@@ -282,3 +282,33 @@ def resolve_host_path(process: Process, ns_path: str) -> str:
     Get a path in the host mount namespace pointing to path in process mount namespace.
     """
     return resolve_proc_root_links(get_proc_root_path(process), ns_path)
+
+
+def get_host_pid(nspid: int, container_id: str) -> Optional[int]:
+    assert len(container_id) == 64, f"Invalid container id {container_id!r}"
+
+    pid_namespace = ""
+    running_processes = list(process_iter())
+
+    # Get the pid namespace of the given container
+    for process in running_processes:
+        try:
+            if container_id in Path(f"/proc/{process.pid}/cgroup").read_text():
+                pid_namespace = os.readlink(f"/proc/{process.pid}/ns/pid")
```

Comment:
You can add an optimization here - 
```code
            if container_id in Path(f"/proc/{process.pid}/cgroup").read_text():
                pid_namespace = os.readlink(f"/proc/{process.pid}/ns/pid")
                if get_process_nspid(process) == nspid:
                    return process.pid
                break
```
WDYT?

Reply:
https://github.com/Granulate/granulate-utils/pull/27/files#r795456428